tools/eslint-rules/utilities/member-expression/types.ts

```diff
@@ -0,0 +1,185 @@
+import { AST_NODE_TYPES, TSESTree } from '@typescript-eslint/utils';
+
+/**
+ * General function that returns MemberExpressions.
+ * Normally first argument is a `TSESTree` node, but could be empty
+ * if function always returns the same result like an empty array.
+ */
+export type MemberExpressionGetter = (
+  ...args: unknown[]
+) => TSESTree.MemberExpression[];
+
+/**
+ * Maps every `AST_NODE_TYPES` to its `TSESTree` equivalent.
```

Comment:
Can you add to this comment to explain when folks should use TESNodes?

Reply:
Removed, so neverrr~~

Removed in favor for the built-in `TSESTree.Node` type